test/solidity/Facets/HopFacetL1Testnet.t.sol

```diff
@@ -0,0 +1,157 @@
+        functionSelectors[0] = hopFacet.startBridgeTokensViaHop.selector;
+        functionSelectors[1] = hopFacet
+            .swapAndStartBridgeTokensViaHop
+            .selector;
+        functionSelectors[2] = hopFacet.initHop.selector;
+        functionSelectors[3] = hopFacet.registerBridge.selector;
+        functionSelectors[4] = hopFacet.addDex.selector;
+        functionSelectors[5] = hopFacet
+            .setFunctionApprovalBySignature
+            .selector;
+
+        addFacet(diamond, address(hopFacet), functionSelectors);
+
+        HopFacet.Config[] memory configs = new HopFacet.Config[](3);
+        configs[0] = HopFacet.Config(ADDRESS_USDC, USDC_BRIDGE);
+        configs[1] = HopFacet.Config(ADDRESS_DAI, DAI_BRIDGE);
+        configs[2] = HopFacet.Config(address(0), NATIVE_BRIDGE);
+
+        hopFacet = TestHopFacet(address(diamond));
+        hopFacet.initHop(configs);
+
+        hopFacet.addDex(address(uniswap));
+        hopFacet.setFunctionApprovalBySignature(
+            uniswap.swapExactTokensForTokens.selector
+        );
+        hopFacet.setFunctionApprovalBySignature(
+            uniswap.swapTokensForExactETH.selector
+        );
+        hopFacet.setFunctionApprovalBySignature(
+            uniswap.swapETHForExactTokens.selector
+        );
+        setFacetAddressInTestBase(address(hopFacet), "HopFacet");
+
+        vm.makePersistent(address(hopFacet));
+
+        // adjust bridgeData
+        bridgeData.bridge = "hop";
+        bridgeData.destinationChainId = DSTCHAIN_ID;
+
+        // produce valid HopData
+        validHopData = HopFacet.HopData({
+            bonderFee: 0,
+            amountOutMin: 0,
+            deadline: block.timestamp + 60 * 20,
+            destinationAmountOutMin: 0,
+            destinationDeadline: block.timestamp + 60 * 20,
+            relayer: address(0),
+            relayerFee: 0,
+            nativeFee: 10000000000000000
+        });
+
+        // native fee expected to be payed on top
+        addToMessageValue = 10000000000000000;
+    }
+
+    function initiateBridgeTxWithFacet(bool isNative) internal override {
+        if (isNative) {
+            // fee parameter Native
+            validHopData.relayerFee = 10000000000000000;
+            validHopData.relayer = 0x81682250D4566B2986A2B33e23e7c52D401B7aB7;
+        } else {
+            // fee parameter ERC20
+            validHopData.relayer = 0xB47dE784aB8702eC35c5eAb225D6f6cE476DdD28;
+        }
+
+        if (isNative) {
+            hopFacet.startBridgeTokensViaHop{
+                value: bridgeData.minAmount + validHopData.nativeFee
+            }(bridgeData, validHopData);
+        } else {
+            hopFacet.startBridgeTokensViaHop{ value: validHopData.nativeFee }(
+                bridgeData,
+                validHopData
+            );
+        }
+    }
+
+    function initiateSwapAndBridgeTxWithFacet(
+        bool isNative
+    ) internal override {
+        if (bridgeData.sendingAssetId == address(0)) {
```

Comment:
Is this supposed to be different here (not checking if "isNative")? If so, would be nice to add a comment to explain the reasoning